src/components/RichTextEditor/Toolbar.tsx

```diff
@@ -74,30 +90,17 @@ export const Toolbar: FC<ToolbarProps> = ({ editorId, textStyles }) => {
                 data-test-id="toolbar"
                 className="tw-flex tw-p-0.5 tw-items-center tw-bg-white tw-rounded tw-shadow-mid tw-gap-0.5"
             >
-                <ButtonGroup testId="text-style-buttons">
-                    <TextStyleDropdown editorId={editorId} textStyles={textStyles} />
-                </ButtonGroup>
-                <ButtonGroup testId="text-alignment-buttons">
-                    <AlignToolbarButton value="left" icon={<IconTextAlignLeft />} />
-                    <AlignToolbarButton value="center" icon={<IconTextAlignCenter />} />
-                    <AlignToolbarButton value="right" icon={<IconTextAlignRight />} />
-                    <AlignToolbarButton value="justify" icon={<IconTextAlignJustify />} />
-                </ButtonGroup>
-
-                <ButtonGroup testId="text-mark-buttons">
-                    <MarkToolbarButton type={getPluginType(editor, MARK_BOLD)} icon={<IconBold />} />
-                    <MarkToolbarButton type={getPluginType(editor, MARK_ITALIC)} icon={<IconItalic />} />
-                    <MarkToolbarButton type={getPluginType(editor, MARK_UNDERLINE)} icon={<IconUnderline />} />
-                    <MarkToolbarButton type={getPluginType(editor, MARK_STRIKETHROUGH)} icon={<IconStrikethrough />} />
-                    <MarkToolbarButton type={getPluginType(editor, MARK_CODE)} icon={<IconSnippet />} />
-                    <BlockToolbarButton type={getPluginType(editor, ELEMENT_CHECK_ITEM)} icon={<IconListChecklist />} />
-                </ButtonGroup>
-
-                <ButtonGroup testId="text-element-buttons">
-                    <LinkToolbarButton icon={<IconLink />} />
-                    <ListToolbarButton type={getPluginType(editor, ELEMENT_UL)} icon={<IconListBullets />} />
-                    <ListToolbarButton type={getPluginType(editor, ELEMENT_OL)} icon={<IconListNumbers />} />
-                </ButtonGroup>
+                {toolbarActions.map((actions, index) => (
+                    <div
+                        key={index}
+                        data-test-id={`toolbar-group-${index}`}
+                        className="tw-flex tw-items-center tw-border-r last:tw-border-r-0 tw-px-3 tw-py-2 tw-border-black-5"
+                    >
+                        {actions.map((action, index) => (
+                            <React.Fragment key={index}>{toolbarComponents[action]}</React.Fragment>
```

Comment:
With the suggested change from above:
```suggestion
                            <React.Fragment key={index}>{toolbarComponents(editorId, textStyles)[action]}</React.Fragment>
```

Reply:
Changed